File: 1003_サイコロの実装(1)/main.py
from abc import ABCMeta, abstractmethod

class IOProvider(metaclass=ABCMeta):
    @abstractmethod
    def input(self) -> str:
        pass
    @abstractmethod
    def output(self, s: str) -> None:
        pass

class StandardIOProvider(IOProvider):
    def input(self) -> str:
        return input()

    def output(self, s: str) -> None:
        print(s)

class TestIOProvider(IOProvider):
    def input(self) -> str:
        return "6"

    def output(self, s: str) -> None:
        print(s)

class Program:
    def __init__(self, provider: IOProvider):
        self.io = provider

    def read(self) -> str:
        return self.io.input()

    def write(self, s: str):
        self.io.output(s)

    def main(self) -> None:
        i = self.read()

        n = int(i)
        l1: list = [x % 6 + 1 for x in range(1, n + 1)]
        match = lambda x, l: list(filter(lambda y: y == x, l))
        l2: list = [len(match(x, l1)) for x in range(1, 7)]
        result: bool = all([ x == l2[0] for x in l2 ])

        self.write("Yes" if result else "No")


def main():
    provider = TestIOProvider()
    Program(provider).main()

if __name__ == '__main__':
    main()

File: 1003_サイコロの実装(1)/main.swift


protocol IOProvider {
  func input() -> String
  func output(_ str: String) -> Void
}

class StandardIOProvider: IOProvider {
  func input() -> String {
    return readLine()!
  }

  func output(_ str: String) -> Void {
    return print(str)
  }
}

class TestIOProvider: IOProvider {
  func input() -> String {
    return "6"
  }

  func output(_ str: String) -> Void {
    print(str)
  }
}

class FileIOProvider: IOProvider {
  func input() -> String {
    return ""
  }

  func output(_ str: String) -> Void {

  }
}

class Base {
  let io: IOProvider;

  init(_ provider: IOProvider) {
    io = provider
  }

  internal func read() -> String {
    return io.input()
  }

  internal func write(_ str: String) -> Void {
    io.output(str)
  }

  public func main() {}
}


class Program: Base {

  public override func main() {
    let input: String = self.read()
    let n:Int? = Int(input)

    if let m = n {
      let l1:[Int] = (1...m).map({ $0 % 6 + 1 })
      let match = { (x) -> [Int] in l1.filter{ $0 == x } }
      let l2:[Int] = (1...6).map({ match($0).count })

      let result:Bool = l2.allSatisfy({ $0 == l2[0] })

      self.write(result ? "Yes" : "No")
    }
  }
}


func main() -> Void {
  let provider = TestIOProvider()
  Program(provider).main()
}

main()